styler v1.1.0 (#12)
lorenzwalthert/blogdown-cactus: content/posts/2018-10-20-styler-1.1.0.Rmd

---
title: styler 1.0.1
author: Lorenz Walthert
date: '2018-10-20'
categories:
  - R
tags:
  - packages
  - styler
description: styler gains syntax highlighting, can style Rnw files, roxygen
  code examples and more in version 1.1.0.
slug: styler-v1.1.0
comments: true
---

[styler](https://github.com/r-lib/styler) 1.1.0 is now available on CRAN. This 
release introduces new features and is fully backward-compatible. It also
adapts to changes in the R parser committed into R devel ([#419](https://github.com/r-lib/styler/issues/419)). Major changes
are:

> * styler can now style roxygen code examples in the source code of package.

Reviewer: ```suggestion
* styler can now style roxygen code examples in the source code of a package.
```

  This might be helfpul for package developers. The option is enabled by
  default, so you have to explicitly turn it off if you want to styler to leave
  the code examples as is: 

```{r, eval = FALSE}
styler::style_pkg(include_roxygen_examples = FALSE)
```
  
  
* In addition to R and Rmd files, styler can now handle Rnw files too. The API 
  ignores capitalization and dot-prefixing for file types:

```{r, eval = FALSE}
styler::style_dir(filetype = c("r", "Rmd", ".rnw"))
```
* the print method for the output of `style_text()` (`print.vertical()`) now
  returns syntax-highlighted code by default, controllable via the option 
  `styler.colored_print.vertical` using the [prettycode](https://CRAN.R-project.org/package=prettycode) package. This only 

>   works when ran in a console though, so you can c/p this snippet into yours

Reviewer: ```suggestion
  works when run in a console though, so you can c/p this snippet into your R console
```

  to see how it works:
  
```{r}
styler::style_text(c("add <- function(...) {", "purrr::reduce(list(...), `+`)", "}"))
```
  

There were also quite a few minor changes, the most visible ones to the user 

> were a complete re-design of the README and the fact that the cursor position is 

Reviewer: ```suggestion
are a complete re-design of the README and the fact that the cursor position is 
```
